recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,510 @@
+        if self.options.with_z3:
+            self.requires("z3/4.13.0")
+
+    def build_requirements(self):
+        self.tool_requires("ninja/1.12.1")
 
     def validate(self):
-        if self.options.shared:  # Shared builds disabled just due to the CI
-            message = 'Shared builds not currently supported'
-            raise ConanInvalidConfiguration(message)
-            # del self.options.fPIC
-        # if self.settings.os == 'Windows' and self.options.shared:
-        #     message = 'Shared builds not supported on Windows'
-        #     raise ConanInvalidConfiguration(message)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.options.shared:
+            if self._is_windows:
+                raise ConanInvalidConfiguration("Shared builds not currently supported on Windows")
+            if os.getenv("CONAN_CENTER_BUILD_SERVICE") and self.settings.build_type == "Debug":
+                raise ConanInvalidConfiguration("Shared Debug build is not supported on CCI due to resource limitations")
+            if is_apple_os(self):
+                # FIXME iconv contains duplicate symbols in the libiconv and libcharset libraries (both of which are
+                #  provided by libiconv). This may be an issue with how conan packages libiconv
+                iconv_dep = self.dependencies.get("libiconv")
+                if iconv_dep and not iconv_dep.options.shared:
+                    raise ConanInvalidConfiguration("Static iconv cannot be linked into a shared library on macos "
+                                                    "due to duplicate symbols. Use libxml2/*:iconv=False.")
+
         if self.options.exceptions and not self.options.rtti:
-            message = 'Cannot enable exceptions without rtti support'
-            raise ConanInvalidConfiguration(message)
-        self._supports_compiler()
-        if cross_building(self, skip_x64_x86=True):
-            raise ConanInvalidConfiguration('Cross-building not implemented')
+            raise ConanInvalidConfiguration("Cannot enable exceptions without rtti support")
 
-    def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True,
-            destination=self._source_subfolder)
-        self._patch_sources()
+        if cross_building(self):
+            # FIXME support cross compilation, at least for common cases like Apple Silicon -> X86
+            #  requires a host-compiled version of llvm-tablegen
+            raise ConanInvalidConfiguration("Cross compilation is not supported")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("Cross compilation is not supported. Contributions are welcome!")
```
Just to let the user know that this is not a limitation of LLVM and could be added in the future.